types/react-native/index.d.ts

```diff
@@ -8634,6 +8634,11 @@ export interface SwitchProps extends SwitchPropsIOS {
      */
     disabled?: boolean;
 
+    /**
+     * Invoked with the the change event as an argument when the value changes.
+     */
+    onChange?: (event?: any) => Promise<void> | void;
```

Comment:
Two remarks based on https://github.com/facebook/react-native/blob/master/Libraries/Components/Switch/Switch.js#L79:
* Can we make the `onChange`-prop not only accept `undefined` but also `null`, as it is a maybe-type in the source?
* Can we type the `SwitchChangeEvent` as well?

And do you mind adding some tests?

Reply:
* I added some tests
* I added `null` accepted type

> Can we type the SwitchChangeEvent as well?

unfortunately react-native doesn't export `SwitchChangeEvent`. So I can to add it to declaration types file, but I can't import this event in my application (or I don't know how to do it) see https://github.com/facebook/react-native/blob/master/Libraries/Components/Switch/Switch.js#L27

Good solution will be:

* add `export SwitchChangeEvent` to react-native repo
* open pull request
* wait to merge
* type SwitchChangeEvent here

But it might spend a lot of time

Reply:
Hey :)

Sorry, I think my first comment was a bit misleading, as the maybe type from flow (https://flow.org/en/docs/types/maybe/) applies to the whole prop (it can be `null`, `undefined` or the callback):
```diff
- onChange?: (event?: any) => Promise<void> | void | null;
+ onChange?: ((event?: any) => Promise<void> | void) null;
```

In regards to the event: You can simply declare the event as interface in the file and use it here, similar additions were done for other events: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/777174d7912b0dbe86fe0081f00c196d626956e8/types/react-native/index.d.ts#L410

Reply:
@ca057 thanks for your advices. I've fixed everything. Look it again please

Reply:
Looks good to me!